Marlin/src/pins/samd/pins_RAMPS_144.h

```diff
@@ -260,79 +275,224 @@
   #define SDSS                                53
 #endif
 
+//
+// Průša i3 MK2 Multiplexer Support
+//
+#if HAS_PRUSA_MMU1
+  #ifndef E_MUX0_PIN
+    #define E_MUX0_PIN                   AUX2_06  // Z_CS_PIN
+  #endif
+  #ifndef E_MUX1_PIN
+    #define E_MUX1_PIN                   AUX2_08  // E0_CS_PIN
+  #endif
+  #ifndef E_MUX2_PIN
+    #define E_MUX2_PIN                   AUX2_07  // E1_CS_PIN
+  #endif
+#endif
+
+//
+// AUX1    5V  GND D2  D1
+//          2   4   6   8
+//          1   3   5   7
+//         5V  GND A3  A4
+//
+#define AUX1_05                               57  // (A3)
+#define AUX1_06                                2
+#define AUX1_07                               58  // (A4)
+#define AUX1_08                                1
+
+//
+// AUX2    GND A9 D40 D42 A11
+//          2   4   6   8  10
+//          1   3   5   7   9
+//         VCC A5 A10 D44 A12
+//
+#define AUX2_03                               59  // (A5)
+#define AUX2_04                               63  // (A9)
+#define AUX2_05                               64  // (A10)
+#define AUX2_06                               40
+#define AUX2_07                               44
+#define AUX2_08                               42
+#define AUX2_09                               66  // (A12)
+#define AUX2_10                               65  // (A11)
+
+//
+// AUX3    GND D52 D50 5V
+//          7   5   3   1
+//          8   6   4   2
+//         NC  D53 D51 D49
+//
```

Comment:
AUX3 in ramps 1.4.4 is a 5x2 connector
  RST GND  SCK MISO VCC
  NC  5V     D53 MOSI D49

because D52, D50 and D51 are not SPI pins, real SPI pins are named SCK, MISO and MOSI and are the same pins used by SPI connector on due (named ICSP on AGCM4)

Reply:
Even if ramps 1.4.4 supports due and then may redirect D52, D50 and D51 on AX3 this pinout is specific for ramps 1.4.4 with AGCM then it requires such changes

Reply:
"RAMPS 1.4.4" by itself is meaningless. It requires a base mapping of the underlying pins that mate up with the shield. The basic `pins_RAMPS.h` is based on a RAMPS mated with a standard Arduino Mega.

Reply:
on this assumptions you have to move it outside samd folder, change board name (AGCM4 RAMPS 1.4.4) and remove:
```
#if NOT_TARGET(ARDUINO_GRAND_CENTRAL_M4)
  #error "Oops! Select 'Adafruit Grand Central M4' in 'Tools > Board.'"
#endif
```

Reply:
but doing this it will be no more useable with an AGCM4. A new pins definition should be done, derived from this, with correct SPI changes

Reply:
> you have to move it outside samd folder

If RAMPS 144 is the only way that AGCM4 can be interfaced with Marlin, then the file might as well be renamed to `pins_AGCM4_RAMPS_144.h`. If AGCM4 always uses the SAMD processor then it should remain in the "samd" folder. If other boards arrive that want to interface with RAMPS 144, they can be named to indicate them as such.

The longstanding use of a "ramps" folder to contain boards that follow the RAMPS pin mappings might just be approaching obsolescence. It has been convenient mainly due to the shared remapping of Extruder / Bed / Fan pins based on configuration. That might also get an additional layer similar to this LCD refactor.

Each of these tasks is a small pile of work, so be patient. I will continue on all of these concepts alonside my regular duties.